tests/unit/test_toolchain.cpp

```diff
@@ -62,16 +70,20 @@ void test_packed_union()
         uint16_t mField;
     } nested_t;
 
-    typedef OT_TOOL_PACKED_BEGIN struct
+    OT_TOOL_PACKED_BEGIN
+    struct packed_t
     {
         uint8_t mBytes[3];
         union
         {
             nested_t mNestedStruct;
             uint8_t  mByte;
         } OT_TOOL_PACKED_FIELD;
-    } OT_TOOL_PACKED_END packed_t;
+    } OT_TOOL_PACKED_END;
 
+#if _WIN32
+    static_assert(sizeof(packed_t) == 5, "Toolchain::OT_TOOL_PACKED failed 3");
+#endif
```

Comment:
If `static_assert` is always a macro, then we could change `#if _WIN32` to `#ifdef static_assert`. Or we could add a `etc/missing/static_assert` to define static asserts for other platforms, since static asserts are pretty useful.